packages/app-store/routing-forms/pages/route-builder/[...appPages].tsx

```diff
@@ -196,16 +195,34 @@ const buildEventsData = ({
       const isRouteAlreadyInUse = isRouter(route) ? false : uniqueSlug === route.action.value;
 
       // If Event is already in use, we let it be so as to not break the existing setup
+
+      // Pass app data that works with routing forms
+      const eventTypeAppMetadataParse = EventTypeAppMetadataSchema.safeParse(eventType.metadata?.apps);
+      const eventTypeAppMetadata: Record<string, any> = {};
+      if (eventTypeAppMetadataParse.success) {
+        const appMetadata = eventTypeAppMetadataParse.data;
+
+        if (appMetadata) {
+          for (const appSlug of Object.keys(appMetadata)) {
+            if (enabledAppSlugs.includes(appSlug)) {
+              eventTypeAppMetadata[appSlug] = appMetadata[appSlug as keyof typeof appMetadata];
+            }
+          }
+        }
+      }
```

Comment:
I know it is in draft but still leaving code feedback.


Let's abstract it out